### PR TITLE
Applied Limit to show reading trends

### PR DESCRIPTION
**Pull request by shariq1501**

### CODE REVIEW

**Comment by MonikaSharma06** [comment]
Getting Error if we enter Offset value:

errorafterenteringoffsetvalue

File: src/frontend/src/app/asset-readings/chart-modal/chart-modal.component.html [hunk 1]
```diff
@@ -7,7 +7,7 @@
           <p class="modal-card-title">{{assetCode}}</p>
         </div>
         <div class="column is-4">
-          <input-debounce placeholder="Limit(max 1000)" #limit (value)="plotReadingsGraph(assetCode, $event, offset.inputValue)" max="100"></input-debounce>
+          <input-debounce placeholder="Limit(max 1000)" #limit (value)="plotReadingsGraph(assetCode, $event, offset.inputValue)" max="1000"></input-debounce>
```
Comment (MonikaSharma06) [inline src/frontend/src/app/asset-readings/chart-modal/chart-modal.component.html]: If I am right, it should not allow user to enter Limit > 1000 and should show an error but, it allows:
noerrorforlimit 1000

**Comment by MonikaSharma06** [comment]
Graph stuck on it's previous image if we enter any Negative (-ve) value in Limit, rather than it should show `No Data` or `Not a Valid Input` in this case: 
negativelimitvalue
positivelimitvalue

**Comment by shariq1501** [comment]
@monika-Dianomic  Fixed invalid input error message issue. Please review now.

**Comment by MonikaSharma06** [comment]
@mshariq-nerd It works fine now except one thing, It accept characters (-, +) with numeric value. Please also add validation for this type of value.

**Comment by shariq1501** [comment]
@monika-Dianomic  All Issue fixed. Please review and approve.

**Comment by MonikaSharma06** [comment]
If we enter invalid Offset, it shows `Not Available` message with no graph. But if we enter invalid Limit, then it shows 'Invalid Input' validation message and shows the default graph for 20 records.

invalidoffset
invalidlimit

File: src/frontend/src/app/asset-readings/chart-modal/chart-modal.component.ts [hunk 2]
```diff
@@ -36,26 +38,41 @@ export class ChartModalComponent implements OnInit {
...
     this.isValidData = true;
+    this.isReadingsAvailable = true;
+    this.isInvalidInput = false;
+    if (limit == null) {
+      limit = 0;
+    }
+    if (offset == null) {
+      offset = 0;
+    }
+    if (limit < 0 || limit > 1000 || offset < 0 || !Number.isInteger(limit)
```
Comment (praveen-garg) [inline src/frontend/src/app/asset-readings/chart-modal/chart-modal.component.ts]: separate limit and offset check!
`if (limit < 0 || limit > 1000 || !Number.isInteger(limit)`
and next if for offset

File: src/frontend/src/app/asset-readings/chart-modal/chart-modal.component.ts [hunk 3]
```diff
@@ -36,26 +38,41 @@ export class ChartModalComponent implements OnInit {
...
           return;
+        } else if (data.length === 0) {
+          this.isReadingsAvailable = false;
+          return;
         }
         const validRecord = ReadingsValidator.validate(data);
         if (validRecord) {
           this.getAssetTimeReading(data);
-           const dataObj = {
+          const dataObj = {
```
Comment (praveen-garg) [inline src/frontend/src/app/asset-readings/chart-modal/chart-modal.component.ts]: x`Obj` ?

File: src/frontend/src/app/asset-readings/chart-modal/chart-modal.component.ts [hunk 4]
```diff
@@ -194,4 +210,9 @@ export class ChartModalComponent implements OnInit {
       datasets: ds
     };
   }
+
+  clearField(limitField, offsetField) {
+    limitField.inputValue = '';
```
Comment (praveen-garg) [inline src/frontend/src/app/asset-readings/chart-modal/chart-modal.component.ts:219]: don;t we have binding of component html > ts fields binding with `this` ?

File: src/frontend/src/app/asset-readings/chart-modal/chart-modal.component.css [hunk 5]
```diff
@@ -0,0 +1,13 @@
+.delete {
```
Comment (praveen-garg) [inline src/frontend/src/app/asset-readings/chart-modal/chart-modal.component.css:1]: is this css common of all modal?

**Review by praveen-garg — CHANGES_REQUESTED** [review]

**Comment by MonikaSharma06** [comment]
It accepts invalid input:

acceptsinvalidinput

File: src/frontend/src/app/asset-readings/chart-modal/chart-modal.component.ts (hunk 4 above)
Comment (shariq1501) [reply src/frontend/src/app/asset-readings/chart-modal/chart-modal.component.ts:219]: that is not required. it's a local field in html tamplate and pass as a parameter in method

**Comment by shariq1501** [comment]
This is because `parseInt` discard values after `-`. 
`Number.parseInt(2-4) => 2`. This is something wired, not sure what is the logic behind this.

File: src/frontend/src/app/asset-readings/chart-modal/chart-modal.component.css (hunk 5 above)
Comment (shariq1501) [reply src/frontend/src/app/asset-readings/chart-modal/chart-modal.component.css:1]: This is only for chart modal.

**Review by praveen-garg — APPROVED** [review]

**Pull request merged**